### PR TITLE
Fix documentation and acceptable values for `filter_pmap`'s 'donate'

### PR DESCRIPTION
**Pull request by colehaus**
Resolves https://github.com/patrick-kidger/equinox/issues/467.

Note that I wasn't able to run most tests because I kept getting segfaults (I assume this is because I'm on NixOS which is kind of a special snowflake and I didn't properly Nixify the project).

### CODE REVIEW

File: equinox/_vmap_pmap.py [hunk 1]
```diff
@@ -698,11 +698,18 @@ def g(x, y):
...
         )
 
-    if donate not in {"arrays", "warn", "none"}:
+    if donate == "arrays":
+        warnings.warn(
+            "The `donate='arrays'` option to `filter_pmap` has been renamed to "
+            "`donate='all'`",
+            DeprecationWarning,
+        )
+        donate == "all"
```
Comment (patrick-kidger) [inline equinox/_vmap_pmap.py:707]: This should be a single `=`.

**Review by patrick-kidger — COMMENTED** [review]
Typo aside, this LGTM -- thank you for the fix, this is really helpful!

**Comment by patrick-kidger** [comment]
And merged! Thank you for the contribution.

**Comment by colehaus** [comment]
Thanks!

**Pull request merged**